### DISCUSSION TITLE
The preload preload failed.

### DISCUSSION THREAD

**Discussion by MYDJH** [iTwin/itwinjs-core#6575] [Q&A] [answered]
On the desktop, I want to enable communication between the webpage and the main process using preload, but I haven't been successful so far. I have looked at the relevant file, ElectronHost.ts.

![image.png](https://p9-juejin.byteimg.com/tos-cn-i-k3u1fbpfcp/737fabd8f0b048f6baad8a44fa7b2fe6~tplv-k3u1fbpfcp-jj-mark:0:0:0:0:q75.image#?w=771&h=297&s=57751&e=png&b=ffffff)

I created a new ElectronPreload.js file next to the entry file, but my application didn't load successfully.
![image.png](https://p9-juejin.byteimg.com/tos-cn-i-k3u1fbpfcp/15a0d19477734d66acaabc41361d6fa8~tplv-k3u1fbpfcp-jj-mark:0:0:0:0:q75.image#?w=1313&h=441&s=271730&e=png&b=1d1d1d)

Do I need to add any other configuration parameters? I hope to receive your assistance. Looking forward to your response.

**Comment by GytisCepk** [comment, marked as answer]
Preloading is not available when using `ElectronHost.openMainWindow(...)`. 

If you're interested in setting up communication between your renderer and main processes, I recommend looking into using iTwin.js IPC. ["Implementing Ipc Interfaces"](https://www.itwinjs.org/learning/ipcinterface/) will explain how to setup communication using your custom IPC Interface and `IpcHandler`. If you still want to set up your own Electron IPC communication, you should try doing it outside preload.

If you have further questions, let us know.

**Reply by MYDJH** [reply]
Thank you very much.
